lang/rs/bindings/cners/src/config.rs

```diff
@@ -0,0 +1,518 @@
+    bufcnt: u32,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+struct Lport<'a> {
+    netdev: Option<String>,
+    pmd: String,
+    qid: u16,
+    umem: String,
+    #[serde(default)]
+    region: usize,
+    #[serde(default)]
+    busy_polling: bool,
+    #[serde(default)]
+    busy_timeout: u16,
+    #[serde(default)]
+    busy_budget: u16,
+    #[serde(default)]
+    unprivileged: bool,
+    #[serde(default)]
+    force_wakeup: bool,
+    #[serde(default)]
+    skb_mode: bool,
+    description: Option<String>,
+    #[serde(skip)]
+    xdp_uds: Option<&'a mut uds_info_t>,
+    #[serde(skip)]
+    lport_id: Option<u16>,
+    #[serde(skip)]
+    xsk_dev: Option<&'a mut xskdev_info_t>,
+}
+
+#[derive(Serialize, Deserialize)]
+struct LcoreGroups {
+    initial: Option<Vec<i32>>,
+    group0: Option<Vec<i32>>,
+    group1: Option<Vec<i32>>,
+    default: Option<Vec<String>>,
+}
+
+#[derive(Serialize, Deserialize)]
+struct Options {
+    pkt_api: Option<String>,
+    #[serde(default, rename = "no-metrics")]
+    no_metrics: bool,
+    #[serde(default, rename = "no-restapi")]
+    no_restapi: bool,
+    #[serde(default)]
+    cli: bool,
+    mode: Option<String>,
+    uds_path: Option<String>,
+}
+
+#[derive(Serialize, Deserialize)]
+struct Thread {
+    group: Option<String>,
+    lport: Option<Vec<String>>,
+    description: Option<String>,
+}
+
+#[derive(Serialize, Deserialize)]
+pub(crate) struct Config<'a> {
+    application: Application,
+    defaults: Option<Defaults>,
+    umems: HashMap<String, UmemData<'a>>,
+    lports: HashMap<String, Lport<'a>>,
+    #[serde(rename = "lcore-groups")]
+    lcore_groups: Option<LcoreGroups>,
+    options: Option<Options>,
+    threads: Option<HashMap<String, Thread>>,
+}
+
+// Implement dummy Debug trait for xskdev_info_t.
+impl std::fmt::Debug for xskdev_info_t {
+    fn fmt(&self, _f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        Ok(())
+    }
+}
+
+impl<'a> Config<'a> {
+    pub fn load_config(jsonc_file: &str) -> Result<Config<'a>, CneError> {
+        let contents = fs::read_to_string(jsonc_file).map_err(|e| {
+            let msg = format!("Error reading file {} ", e);
+            CneError::ConfigError(msg)
+        })?;
+        let res = serde_json::from_str::<Config>(&contents).map_err(|e| {
+            let msg = format!("serde json error {} ", e);
+            CneError::ConfigError(msg)
+        });
+        res
+    }
+
+    #[cfg(test)]
+    pub fn get_config(cfg: &Config) -> Result<String, CneError> {
+        serde_json::to_string(cfg).map_err(|e| CneError::ConfigError(e.to_string()))
+    }
+
+    pub fn setup(&mut self) -> Result<(), CneError> {
+        self.setup_mem_pool()?;
+
+        self.setup_lports()?;
+
+        Ok(())
+    }
+
+    pub fn cleanup(&mut self) -> Result<(), CneError> {
+        self.cleanup_lports()?;
+
+        self.cleanup_mem_pool()?;
+
+        Ok(())
+    }
+
+    pub fn get_port_by_name(&self, port_name: &str) -> Result<Port, CneError> {
```

Comment:
Not sure we need get_port_by_name() if we are using port number. something like get_port_info(int port_id) would be more useful IMO.

Reply:
It's an option. We can have both IMO.  I feel using port name in JSON file is more user friendly than counting the port index in lport section in JSON.